### ISSUE TITLE
Easyconfig Dump method not robust enough

### ISSUE THREAD

**Issue by ocaisa** [hpcugent/easybuild-framework#1305]
The [easyconfig dump() method](https://github.com/hpcugent/easybuild-framework/blob/d1387c5847ef3e6d4ffe4c3fba2a0eb8e02ebb46/easybuild/framework/easyconfig/easyconfig.py#L475) is not robust enough to handle easyconfigs which include python code (such as for [libint](https://github.com/hpcugent/easybuild-easyconfigs/blob/e2de9e479830843a2644c08189edadb039f42d5e/easybuild/easyconfigs/l/Libint/Libint-1.1.4-intel-2015a.eb))

**Comment by boegel** [comment]
Any error traceback to share?

**Comment by ocaisa** [comment]
```
  File "/home/alanc/EasyBuild_Git/easybuild-framework/easybuild/framework/easyconfig/easyconfig.py", line 508, in dump
    ebtxt.append("%s = %s" % (key1, to_str(val)))
  File "/home/alanc/EasyBuild_Git/easybuild-framework/easybuild/framework/easyconfig/easyconfig.py", line 485, in to_str
    return "%s" % x
TypeError: not all arguments converted during string formatting```

**Comment by boegel** [comment]
@caylo: please take a look at this

**Comment by boegel** [comment]
@caylo: to make sure `bump` works, and keeps working, after fixing the issue reported here, we should consider including a check for it in https://github.com/hpcugent/easybuild-easyconfigs/blob/master/test/easyconfigs/easyconfigs.py

**Comment by boegel** [comment]
@caylo: to reproduce:

```
$ python
Python 2.7.6 (default, Sep  9 2014, 15:04:36) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.39)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from easybuild.framework.easyconfig.easyconfig import EasyConfig
>>> from test.framework.utilities import init_config
Couldn't import dot_parser, loading of dot files will not be possible.
>>> cfg = init_config()
>>> ec = EasyConfig('/Users/kehoste/work/easybuild-easyconfigs/easybuild/easyconfigs/l/Libint/Libint-1.1.4-goolf-1.4.10.eb')
>>> ec.dump('test.eb')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "easybuild/framework/easyconfig/easyconfig.py", line 508, in dump
    ebtxt.append("%s = %s" % (key1, to_str(val)))
  File "easybuild/framework/easyconfig/easyconfig.py", line 485, in to_str
    return "%s" % x
TypeError: not all arguments converted during string formatting
```

**Comment by boegel** [comment]
fixed by #1308, `dump()` is now also tested on all easyconfigs via https://github.com/hpcugent/easybuild-easyconfigs/pull/1761
